src/network/networkd-dhcp-static-lease.c

```diff
@@ -0,0 +1,231 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "alloc-util.h"
+#include "ether-addr-util.h"
+#include "hashmap.h"
+#include "memory-util.h"
+#include "networkd-dhcp-static-lease.h"
+#include "networkd-util.h"
+#include "strv.h"
+#include "utf8.h"
+
+DHCPStaticLease *dhcp_static_lease_free(DHCPStaticLease *static_lease) {
+        if (!static_lease)
+                return NULL;
+
+        if (static_lease->network && static_lease->section)
+                ordered_hashmap_remove(static_lease->network->dhcp_static_leases_by_section, static_lease->section);
+
+        network_config_section_free(static_lease->section);
+        static_lease->static_lease = sd_dhcp_static_lease_unref(static_lease->static_lease);
+        return mfree(static_lease);
+}
+
+int dhcp_static_lease_new(DHCPStaticLease **ret) {
+        DHCPStaticLease *p = NULL;
+        int r;
+
+        assert(ret);
+
+        p = new0(DHCPStaticLease, 1);
+        if (!p)
+                return -ENOMEM;
+
+        r = sd_dhcp_static_lease_new(&p->static_lease);
+        if (r < 0)
+                return r;
+
+        *ret = TAKE_PTR(p);
+        return 0;
+}
+
+int dhcp_static_lease_set_client_id(DHCPStaticLease *lease, const struct ether_addr *mac_addr) {
+        assert(lease);
+        assert(mac_addr);
+
+        return sd_dhcp_static_lease_set_client_id_by_mac(lease->static_lease, mac_addr->ether_addr_octet, ETH_ALEN);
+}
+
+int dhcp_static_lease_set_address(DHCPStaticLease *lease, const struct in_addr *address) {
+        assert(address);
+        assert(lease);
+
+        return sd_dhcp_static_lease_set_address(lease->static_lease, address);
+}
+
+static int dhcp_static_lease_section_verify(DHCPStaticLease *l) {
+        assert(l);
+
+        if (section_is_invalid(l->section))
+                return -EINVAL;
+
+        assert(l->section);
+
+        return 0;
+}
```

Comment:
This function seems not necessary. Please merge this with `network_drop_invalid_static_leases()`.